### ISSUE TITLE
Adjustments to `plot_access_histogram()` for Darshan Summary Report

### ISSUE THREAD

**Issue by nawtrey** [darshan-hpc/darshan#577]
`plot_access_histogram` needs some final adjustments before it can be added to the Darshan Summary Report. I'll show some specific examples that I think are representative of some of the issues I'm seeing. 

## `noposixopens.darshan`:
![noposixopens darshan_POSIX](https://user-images.githubusercontent.com/46797896/144533954-3dd875eb-303a-4720-b623-e5ba7e0f6ed8.png)

The title has a misspelled word, but I think we will remove it like we have for other figures and just add an x-axis label instead. Based on prior conversations about other figures, I think we want to display values even if they are zero, but maybe setting the minimum y values will make this look a little cleaner.

## `theta-large-2021-bz2.darshan`:
![theta-large-2021-bz2 darshan_POSIX](https://user-images.githubusercontent.com/46797896/144533925-17cf67a0-ece8-414c-86a8-180f7e436d3f.png)

This one has some label overlap, and the highest bar label is slightly touching the top of the figure border. I'm not sure the best way to handle this. The best looking solution I've seen would be to make this a horizontal bar graph, and find a way to set the x-axis limits such that the labels never go outside of the figure border. Or maybe just removing the bar labels entirely would be fine, I'm not sure. It depends on what the team wants to convey with these plots. 

In summary:
 - [x] remove the title and add "access size" label to x-axis
 - [ ] set the y minimum so we never see negative tick values
 - [ ] set the y max so the labels are never touching the frame
 - [ ] switch to horizontal bar graphs

I think these cover the primary issues I've seen with the figures. I've listed some other changes that are more code-focused below:
- [x] `autolabel` function should be moved outside of plotting function to allow for testing
- [ ] Add type hints
- [ ] Update documentation

**Comment by nawtrey** [comment]
After feedback from the team it seems we want to switch to horizontal bars. I've updated the checklist to reflect this.

**Comment by shanedsnyder** [comment]
Tying up some loose ends on old issues, and I think the remaining items listed here are either handled in #907 (ylim of 0, removal of right/top axis spines to avoid annotation overlap) or are things we aren't going to do (horizontal orientation of the bars).
